examples/config.kinesis.minimal.hocon

```diff
@@ -0,0 +1,30 @@
+# The minimum required config options for loading from kinesis
+"input": {
+  "type": "kinesis"
+  "streamName": "test-kinesis-stream"
+  "initialPosition": "LATEST"
+  "region": "eu-central-1"
```

Comment:
This is an interesting case. Most of (all?) other apps allow to set the region only once, somewhere at the top level. At the same time, there are different trade-offs and perhaps you did it right and I was doing it wrong.

The trade-offs as I see them, with single region - pros: we can be very concise; we encourage to use the same region; cons: if someone is stuck with different regions - they're completely out of luck to use the asset. For individual regions its vice versa - it's not a "minimal" config anymore if we have to repeat the region 3-4 times. Although many regions is purely UX issue, whereas not being able to use with different regions is more like a fundamental flaw.

I'm wondering though if there are any trade-off options we can explore, like adding a single top-level region and encouraging to use it. At the same time providing an ability to use per-service/stream regions. Latter always have priority over former.

Reply:
Didn't think about this before but I like the last approach that you suggested. I think we can go with that one.

Reply:
I suggest:

Continue to allow multi-region.  Don't have a top-level `region` field.  Instead, make `region` optional everywhere.  If it is unset then use the `DefaultCredentialsProviderChain`.

This allows the minimal hocon to be very simple (no regions!), which will probably cover the majority of deployments.  It allows power users the fine-grained control they need.  And the code is relatively simple because region is only set at one level.

Reply:
Just to give heads up, we are using [region to create Kinesis endpoint](https://github.com/snowplow/snowplow-elasticsearch-loader/pull/203/files#diff-062d73c4a4e2a587b455f5b2a5be3ec957a5406838154fcfa73bf564e811ba94R92). We need to find a way to creating it without region or we might completely eliminate custom Kinesis endpoint option. I can understand the use case for it but I am not sure how much used that one.